File: sdk/basin-java/src/main/java/io/basin/sdk/ApiKeyDescriptor.java
package io.basin.sdk;

import com.fasterxml.jackson.annotation.JsonProperty;

/** One element of the {@code GET /auth/v1/api-keys} array. */
public final class ApiKeyDescriptor {

    @JsonProperty("id")
    public final long id;

    @JsonProperty("name")
    public final String name;

    @JsonProperty("created_at")
    public final String createdAt;

    /** {@code null} when the key has never been used. */
    @JsonProperty("last_used_at")
    public final String lastUsedAt;

    /** {@code null} when the key is still active. */
    @JsonProperty("revoked_at")
    public final String revokedAt;

    public ApiKeyDescriptor(
            @JsonProperty("id") long id,
            @JsonProperty("name") String name,
            @JsonProperty("created_at") String createdAt,
            @JsonProperty("last_used_at") String lastUsedAt,
            @JsonProperty("revoked_at") String revokedAt) {
        this.id = id;
        this.name = name;
        this.createdAt = createdAt;
        this.lastUsedAt = lastUsedAt;
        this.revokedAt = revokedAt;
    }
}

File: sdk/basin-php/src/Basin/Types/ApiKeyDescriptor.php
<?php

declare(strict_types=1);

namespace Basin\Types;

/** GET /auth/v1/api-keys array element. */
final class ApiKeyDescriptor
{
    public function __construct(
        public readonly int $id,
        public readonly string $name,
        public readonly string $createdAt,
        public readonly ?string $lastUsedAt,
        public readonly ?string $revokedAt,
    ) {
    }

    /** @param array<string,mixed> $data */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (int) $data['id'],
            name: (string) $data['name'],
            createdAt: (string) $data['created_at'],
            lastUsedAt: isset($data['last_used_at']) ? (string) $data['last_used_at'] : null,
            revokedAt: isset($data['revoked_at']) ? (string) $data['revoked_at'] : null,
        );
    }
}
